app/controllers/ApplicationController.java

```diff
@@ -0,0 +1,113 @@
+package controllers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import java.util.Set;
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+
+import annotations.interfaces.RequireJson;
+import models.Application;
+import util.Security;
+import play.libs.F.Function;
+import play.libs.F.Promise;
+import play.libs.Json;
+import play.libs.ws.WSClient;
+import play.libs.ws.WSRequest;
+import play.libs.ws.WSResponse;
+import play.mvc.Controller;
+import play.mvc.Http;
+import play.mvc.Result;
+
+public class ApplicationController extends Controller {
+
+    Config config = ConfigFactory.load();
+
+    @Inject WSClient ws;
+
+    @RequireJson() public Promise<Result> createApplication() {
+        JsonNode jsonNode = request().body().asJson();
+        if (!jsonNode.isObject()) {
+            return Promise.pure(badRequest("Json sent was not a Json objects"));
```

Comment:
I've switched away from using JSON. I'm using Play's built in tempting system instead of making an Angular app, so no custom ajax calls to the server are needed.